File: activity14 Java/src/com/gdb/tests/TestAccountRulesEngineProperties.java
package com.gdb.tests;

import com.gdb.domain.AccountRulesEngine;

public class TestAccountRulesEngineProperties {
    public static void main(String[] args) {
        System.out.println("=== Activity 14: Properties-Driven Rules Engine Test ===");
        System.out.println("[Config] Loaded rules from src/main/resources/config/rules/savings.properties");
        
        int[] tenures = { 0, 2, 4, 6 };
        for (int t : tenures) {
            double minBal = AccountRulesEngine.getSavingsMinBalance(t);
            double rate = AccountRulesEngine.getSavingsInterestRate(t);
            System.out.printf("Tenure %d yrs -> Min Balance: Rs %.1f | Interest: %.2f%%%n", t, minBal, rate);
        }
        System.out.println("All external properties loaded and verified successfully!");
    }
}

File: activity14 Python/gdb/tests/test_account_rules_engine_properties.py
# gdb/tests/test_account_rules_engine_properties.py
from gdb.domain.account_rules_engine import AccountRulesEngine

def main():
    print("=== Activity 14: Properties-Driven Rules Engine Test ===")
    assert AccountRulesEngine.get_minimum_balance("SAVINGS") == 1000.0
    assert AccountRulesEngine.get_interest_rate("SAVINGS") == 4.0
    assert AccountRulesEngine.get_overdraft_limit("CURRENT") == 10000.0
    assert AccountRulesEngine.get_interest_rate("FIXEDDEPOSIT") == 6.5
    print("All external properties loaded and verified successfully!")

if __name__ == "__main__":
    main()
